### PR TITLE
Added ORM adapter layer to allow use of Sequel

### PR DESCRIPTION
**Pull request by hlascelles**
ActiveRecord should not be a requirement to use this que-scheduler.

### CODE REVIEW

File: lib/que/scheduler.rb [hunk 1]
```diff
@@ -1,3 +1,4 @@
 require 'que/scheduler/version'
 require 'que/scheduler/scheduler_job'
+require 'que/scheduler/adapters/orm'
 require 'que/scheduler/engine' if Gem::Specification.find_by_name('railties')
```
Comment (m-o-e) [inline lib/que/scheduler.rb]: This line raises a `Gem::MissingSpecError` for me:

```
~/.rbenv/versions/2.5.0/lib/ruby/2.5.0/rubygems/dependency.rb:310:in `to_specs': Could not find 'railties' (>= 0) among 182 total gem(s) (Gem::MissingSpecError)
```

Comment (m-o-e) [inline lib/que/scheduler.rb]: This line raises a `Gem::MissingSpecError` for me:

```
~/.rbenv/versions/2.5.0/lib/ruby/2.5.0/rubygems/dependency.rb:310:in `to_specs': Could not find 'railties' (>= 0) among 182 total gem(s) (Gem::MissingSpecError)
```

**Comment by m-o-e** [comment]
Now I get two different errors.

This code:
```
  require 'que'
  require 'que/scheduler'
  Que.connection = DB
```
gives me:

```
../que-0.14.2/lib/que.rb:64:in `adapter': Que connection not established! (RuntimeError)
```
----

When I change it to:
```
  require 'que'
  Que.connection = DB
  require 'que/scheduler'
```

then I get:
```
Gem::MissingSpecError: Could not find 'activerecord' (>= 0) among 182 total gem(s)
```

**Comment by hlascelles** [comment]
I can't immediately replicate that... Can I get the stacktrace and the Gemfile?

**Comment by m-o-e** [comment]
Sorry for the confusion, I had a bundler mixup and was testing the wrong version.

So, after some more hacking I got your last version to the point of a new error message.
With [this patch](https://gist.github.com/m-o-e/dccd4ac947f3585e0eecdcca2e0df45b) applied, I now get the following:

```json
{
  "lib": "que",
  "event": "job_errored",
  "job": {
    "queue": "",
    "priority": 0,
    "run_at": "2018-01-18 11:32:14 +0100",
    "job_id": 3893,
    "job_class": "Que::Scheduler::SchedulerJob",
    "args": [
      {
        "last_run_time": "2018-01-18T11:25:10+01:00",
        "job_dictionary": [
          "MyJob"
        ]
      }
    ],
    "error_count": 5
  },
  "error": {
    "class": "KeyError",
    "message": "key not found: :last_run_time"
  }
}
```

Not sure where this originates from.

**Comment by hlascelles** [comment]
OK... I've added an explicit require for the `Time.zone` and defensively checked the options being passed for key type. 

Can you try it now?

NB I've not changed the way `count` works yet, as I can't recreate that part. What version of Sequel are you using?

**Comment by hlascelles** [comment]
I've also added  a mutator method to the orm adapter so you can try yours without patching. 

Use `Que::Scheduler::Adapters::Orm.instance = SomeClass.new`

**Comment by hlascelles** [comment]
I've changed the count handler to cater for your patch results. Will cut a gem now.

**Comment by hlascelles** [comment]
Released 1.0.2

**Pull request merged**